File: readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>


int main(void)
{
    string story = get_string("Text: ");

    // declare variables
    float letters = 0;
    float words = 1;
    float sentences = 0;
    int result = 0;

    // array of all letters in upper and lower case
    char alphabet[] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'};

    // for letters/symbols/spaces in story + for letters in alphabet
    for (int l = 0; l < strlen(story); l++)
    {
        for (int i = 0; i < strlen(alphabet); i++)
        {
            // checks if letters in the story match up with the lettersin the alphabet. If yes = we increase number of letters
            if (story[l] == alphabet[i])
            {
                letters++;
            }
        }

        result = isspace(story[l]);

        // checks if we are currently dealing with a space. If yes = we increase word count since a space indicates a word has just been completed.
        if (result != 0)
        {
            words++;
        }

        // checks if there is a new sentence. increases number of sentences
        if (story[l] == '!' || story[l] == '?' || story[l] == '.')
        {
            sentences++;
        }
    }

    // average sentences and letters
    float averagesentences = roundf(((sentences / words) * 100) * 100) / 100;
    float averageletters = roundf(((letters / words) * 100) * 100) / 100;

    // Coleman-Liau index
    int index = roundf(0.0588 * averageletters - 0.296 * averagesentences - 15.8);

    // prints the different reading levels
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
import math


def readability():
    text = input("Text: ")

    # IMPORTANT: set word counter = 1 since we don't account for the last word in the sentence since there are no more white spaces afterwards.
    letter_counter = 0
    word_counter = 1
    sentence_counter = 0

    for letter in text:

        if letter.isalpha() == True:
            letter_counter += 1

        if letter.isspace() == True:
            word_counter += 1

        if letter == "." or letter == "!" or letter == "?":
            sentence_counter += 1

    average_letters = float(letter_counter / word_counter * 100)
    average_sentences = float(sentence_counter / word_counter * 100)

    index = round(0.0588 * average_letters - 0.296 * average_sentences - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


readability()
